File: MK/10월 2주차/귤고르기.java
/**
 * 귤고르기
 * 문제: https://school.programmers.co.kr/learn/courses/30/lessons/138476
 */
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> map = new HashMap<>();
        int ans=0;
        for (int i: tangerine) {
            map.put(i, map.getOrDefault(i,0)+1);
        }

        List<int[]> arr = new ArrayList<>();

        for (Map.Entry<Integer, Integer> entry:map.entrySet()) {
            arr.add(new int[]{entry.getValue(), entry.getKey()});
        }

        Collections.sort(arr, (x,y)->y[0]-x[0]);


        for (int[] tmp : arr) {
            k -= tmp[0];
            ans++;
            if (k<=0) {
                break;
            }
        }

        return ans;
    }
}

File: BW/10월3주차/귤고르기.js
function solution(k, tangerine) {    
    var arr = [];
    for(let i of tangerine){
        arr[i] = (arr[i] || 0)+1;
    }
    arr[0] = 0 // 0에는 빈 값이 들어감 (k는 1이상)
    
    arr.sort((a,b) => b-a); // 내림차순
    
    var cnt=0; // 반복 횟수
    var num=0; // 크기를 누적해서 더하기
    
    for(let i of arr){
        if(num >= k) break;
        num += i;
        cnt++;
    }
    return cnt;
}